### ISSUE TITLE
Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by abdulkareemnalband** [buehler/typescript-hero#292]
<thank you for using and contributing to typescript hero>
<please fill the information below to ease up the process>
<you may even implement it and send a PR which would be very appreciated>

#### System information

- OS:  Windows_NT x64 10.0.15063
- TypeScript Hero version: 1.4.2
- VSCode version: Code 1.16.1 (27492b6bf3acb0775d82d2f87b25a93490673c6d, 2017-09-14T16:38:23.027Z)

#### Description

<what is going wrong>
![image](https://user-images.githubusercontent.com/3139353/30468018-51ec0c72-9a06-11e7-862a-0c8e3171288a.png)

##### developer console log

[-1505453739007.txt](https://github.com/buehler/typescript-hero/files/1305376/-1505453739007.txt)

**Comment by buehler** [comment]
What are you doing when the error happens?

**Comment by micksta72** [comment]
for me error happens when saving a file

**Comment by mackelito** [comment]
I get this error to.. when opening a file

**Comment by gombosg** [comment]
Same here.

**Comment by malorus** [comment]
@buehler  I tried to figure out why the message appears and have made the following observation. I have a simplfied function:

```typescript
    onAnswer(isCorrect: boolean) {           
        this.isCorrect = isCorrect;
        let test: string; 
    }
```

When I now exchange the two lines within this function, I get the error:

```typescript
    onAnswer(isCorrect: boolean) {           
        let test: string; 
        this.isCorrect = isCorrect;
    }
```

Somewhat absurd. When I comment out the `let` declaration, the error disappears again. Or when I change `let` to `var`, then the error disappears as well. 

The same behaviour I have found in a different file as well. This is always in conjunction with the keyword `let`. When changing to var, there is no error anymore.

**Comment by Meligy** [comment]
I think I'm getting it also when `let` is the first line in a function.

Like, in an Angular test file:

```ts
describe('HttpTimeoutInterceptor', () => {
  let httpMock: HttpTestingController;
  ...
});
```

It seems to go away if I change the `let` to `var`, unless I'm imagining, because then my tslint setting changes it back to `let` and I get the error after that.

**Comment by buehler** [comment]
That's a strange error indeed. Are you on macOS, linux or windows?

**Comment by abdulkareemnalband** [comment]
I'm on windows with vscode 1.16 64bit version

**Comment by mackelito** [comment]
I´m on macOS

**Comment by m3l7** [comment]
same error.
vscode 1.16
typescript 2.3.4
arch linux

it's happening only on a specific project, not sure why

**Comment by malorus** [comment]
**new observation**:
Deactivating Typescript hero helps avoiding the error. The very same is true for TSLint. Deactivating TSLint instead of Typescript hero solves it too. After reavtivating it the error won't appear again until you restart visualstudiocode. 

I'm using:
vsc: 1.16.1
typescript: 2.4.2
TSLint: 1.0.7
OS: windows 7/64 bit

**Comment by lucasbasquerotto** [comment]
I thought this was a VSCode issue, but then I disabled my extensions and discovered that if I disabled only ts-hero the error doesn't happen. This is an example that generates the error:

```typescript
export class TestClass {	
	public test() {
		let a = <T>() => { let b = null; };
	} 
}
```

If I remove `<T>` or change `let b` with `var b` the issue doesn't happen, but if I change `let a` with `var a` the issue still happen.

Very weird...

OS: Windows 7 Ultimate x64
VSCode: 1.16.1
Node: 7.9.0
Architeture (of VSCode): x64 (changed recently from x32 to x64)
Typescript: 2.5.2 (also happens in 2.3.4)

This issue is recent, probably because of one of the changes of version 1.4.1 (2017-09-11) or 1.4.2 (2017-09-13).
